test/lib/meetup_test.rb

```diff
@@ -0,0 +1,90 @@
+require 'test_helper'
+require_relative '../support/meetups/sample_api_responses'
+
+class MeetupTest < ActiveSupport::TestCase
+
+#The code is structured so /groups is always called before /events. Both responses must be stubbed.
+#Stub data is dependent on this urlname, do not change
+  setup do
+    Event.delete_all
+    Meetup.stubs(:get).with("/pro/operationcode/groups", options).returns(group_endpoint_response)
+    @urlname = "Operation-Code-Hampton-Roads"
+    Meetup.stubs(:get).with("/#{@urlname}/events", options).returns(event_endpoint_response)
+    @event1 = {:source_id=>"243869029",
+      :source_updated=>Date.today,
+      :name=>"Code and Coffee",
+      :description=>
+       "<p>Have you always wanted to try programming? Are you just getting started and have questions? Or maybe you have some experience under your belt and want to chat with people that get what you're talking about.</p> <p>Join us at Between Friends Coffeshop &amp; Cafe! Drinks and treats provided! Make sure to RSVP so we how many people to prepare for :)</p> ",
+      :url=>"https://www.meetup.com/MilSpouseCoders-Robins-AFB/events/243869029/",
+      :start_date=>Date.today,
+      :end_date=>Date.today,
+      :address1=>"1080 Georgia 96",
+      :address2=>nil,
+      :city=>"Warner Robins",
+      :state=>nil,
+      :zip=>nil,
+      :group=>"MilSpouseCoders - Robins AFB",
+      :source_type=>"Meetup"
+    }
+
+  end
+
+  #response.code defaults to 200 in stub data
+  test "group response code of 200 returns response" do
+    assert_nothing_raised { Meetup.new.operationcode_data }
+  end
+
+  test "group response code not equal to 200 raises error" do
+  	Meetup.stubs(:get).with("/pro/operationcode/groups", options).returns(build_response(400))
+  	assert_raises(Exception) { Meetup.new.operationcode_data }
+  end
+
+  test "event response code of 200 returns response" do
+  	assert_nothing_raised { Meetup.new.get_events_for(@urlname) }
+  end
+
+  test "event response code not equal to 200 raises error" do
+  	Meetup.stubs(:get).with("/#{@urlname}/events", options).returns(build_response(400))
+  	assert_raises( Exception ) { Meetup.new.get_events_for(urlname) }
+  end
+
+  #There are 4 events in the stub data, 1 with no venue
```

Comment:
All the tests from here on are throwing an error because there is an issue with the group_names which is called from event_details_by_group.  The groups are returning as nil.  
It runs fine in the rake task.  
As far as I can see, the difference is in def group_names. When running the rake task, operationcode_data returns a hash "name"=>"Come Learn To Code!".  When this data is mapped using "urlname", the groups render fine.  When the data is mapped while running the test, the groups map as nil. When I use byebug to inspect this during the test, operationcode_data is a hash of the form :name => "Come Learn To Code!".  I copied the stub data directly from the console, so I don't know where the alternate hash syntax came from. Also, does it matter?